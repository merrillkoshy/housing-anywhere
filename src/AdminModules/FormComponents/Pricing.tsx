import React, { useEffect, useState } from "react";
import { Form } from "react-bootstrap";
import data from "../../components/country-currency.json";

const Pricing = ({
	price,
	setPrice,
	currencyCode,
	setCurrencyCode,
}: {
	price: number;
	setPrice: React.Dispatch<React.SetStateAction<number>>;
	currencyCode: string;
	setCurrencyCode: React.Dispatch<React.SetStateAction<string>>;
}) => {
	const [filteredArray, setFilteredArray] = useState<any>([]);
	const [panel, openPanel] = useState(false);

	useEffect(() => {
		return () => {
			setFilteredArray([]);
			openPanel(false);
		};
	}, []);
	const checkVal = () => {
		setFilteredArray(
			data.filter((dt: any) => {
				if (dt.code?.toLowerCase().includes(currencyCode)) {
					return dt;
				}
				return "";
			})
		);
	};
	return (
		<>
			<Form.Group className="mb-3">
				<Form.Label>Price</Form.Label>
				<Form.Control
					type="number"
					min={100}
					placeholder="Enter Price"
					value={price}
					id="price"
					required
					onChange={(e) => {
						e.preventDefault();
						setPrice((e.currentTarget as HTMLInputElement).valueAsNumber);
					}}
				/>
			</Form.Group>

			<Form.Group className="mb-3">
				<Form.Label>Currency Code</Form.Label>
				<Form.Control
					type="text"
					id="currency"
					value={currencyCode}
					required
					onKeyPress={checkVal}
					onChange={(e) => {
						e.preventDefault();
						if (e.currentTarget.value !== "") openPanel(true);
						setCurrencyCode(e.currentTarget.value);
					}}
					placeholder="EUR"
				/>
				<div className={`d-${panel ? "flex" : "none"} autocomplete-items`}>
					{filteredArray.length
						? filteredArray.map((ar: any, i: number) => {
								return (
									<div
										key={ar.code + i + ar.Currency}
										onClick={() => {
											setCurrencyCode(ar.code);
											setFilteredArray([]);
											openPanel(false);
										}}
									>
										{new Set(ar.code)}
									</div>
								);
						  })
						: null}
				</div>
			</Form.Group>
		</>
	);
};
export default Pricing;
