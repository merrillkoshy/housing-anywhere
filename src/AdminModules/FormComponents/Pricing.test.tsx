import React, { SetStateAction } from "react";
import { shallow } from "enzyme";
import Pricing from "./Pricing";

describe("Pricing module", () => {
	const PricingModule = shallow(
		<Pricing
			price={0}
			setPrice={function(value: SetStateAction<number>): void {
				throw new Error("Function not implemented.");
			}}
			currencyCode={""}
			setCurrencyCode={function(value: SetStateAction<string>): void {
				throw new Error("Function not implemented.");
			}}
		/>
	);
	it("renders", () => {
		expect(PricingModule).toMatchSnapshot();
	});
	it("finds a price input which is a number", () => {
		expect(PricingModule.find("#price").exists()).toBe(true);
	});
	it("finds a currency input which is a string", () => {
		expect(PricingModule.find("#currency").exists()).toBe(true);
	});
});
