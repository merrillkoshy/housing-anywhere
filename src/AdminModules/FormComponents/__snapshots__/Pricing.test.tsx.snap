// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pricing module renders 1`] = `
<Fragment>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Price
    </FormLabel>
    <FormControl
      id="price"
      min={100}
      onChange={[Function]}
      placeholder="Enter Price"
      required={true}
      type="number"
      value={0}
    />
  </FormGroup>
  <FormGroup
    className="mb-3"
  >
    <FormLabel
      column={false}
      srOnly={false}
    >
      Currency Code
    </FormLabel>
    <FormControl
      id="currency"
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="EUR"
      required={true}
      type="text"
      value=""
    />
    <div
      className="d-none autocomplete-items"
    />
  </FormGroup>
</Fragment>
`;
